R/utils_memoised.R

```diff
@@ -1,28 +1,24 @@
-
-memoise_cran_db <- memoise::memoise({
-  function() {
-    cran <- tools::CRAN_package_db()
-    # remove first instance of column name MD5Sum
-    cran
-  }
-}
-)
-
-
-
 #' Fetch CRAN Mirrors Info
 #'
 #' @param all default \code{TRUE}, passed to \code{\link{utils}[getCRANmirrors]}
-#' @param host_exists whether cran can be contacted
 #' @param ... additional arguments passed to \code{\link{utils}[getCRANmirrors]}
+#' @param .local an optional local directory to source the CRAN package index
+#'   from, defaulting to \code{getOption("riskmetric.tests")}, used
+#'   for isolating repository requests during testing.
 #'
 #' @importFrom curl nslookup
+#' @importFrom memoise memoise
+#'
 memoise_cran_mirrors <- memoise::memoise({
   # add parameter such that memoised results rerun if internet availability changes
   # NOTE: might need to implement actual caching to avoid inconsistent behavior
   # when run with spotty internet
-  function(all = TRUE, ...,
-           host_exists = !is.null(curl::nslookup("cran.r-project.org", error = FALSE))) {
+  function(all = TRUE, ..., .local = getOption("riskmetric.tests")) {
```

Comment:
suggest to reverse the order of `...` and `.local`. 
suggest to avoid argument start with `.`

Reply:
The rationale for this is that it's only ever used for testing purposes, so the leading `.` is an indication that the parameter isn't intended for most users (in this case, most developers of the package).

With that perspective in mind, do you still think the order and naming should be changed?

Reply:
I see. It goods for me. 

In addition, would the argument `e` required ?

https://github.com/pharmaR/riskmetric/blob/127080d9f4f7d45a6d140d4ee0083d3e7a4521be/R/utils_memoised.R#L25

Reply:
Yes - it at least needs to be a function that accepts one (or more) arguments. The name isn't meaningful, especially when we just suppress it by returning `NULL`. If an issue is caught during evaluation of the expression in `tryCatch`, this function is called before throwing the error. This is how we suppress the output. As a convention, I usually use `m`, `w` and `e` for messages, warnings and errors respectively:

```r
tryCatch(<some expression>
  message = function(m) <handle messages>,
  warning = function(w) <handle warnings>,
  error = function(e) <handle errors>
)
```